integration-cli/docker_cli_run_unix_test.go

```diff
@@ -205,13 +205,37 @@ func (s *DockerSuite) TestRunWithCpuPeriod(c *check.C) {
 func (s *DockerSuite) TestRunWithKernelMemory(c *check.C) {
 	testRequires(c, kernelMemorySupport)
 
-	dockerCmd(c, "run", "--kernel-memory", "50M", "--name", "test", "busybox", "true")
+	dockerCmd(c, "run", "--kernel-memory", "50M", "--name", "test1", "busybox", "true")
 
-	out, err := inspectField("test", "HostConfig.KernelMemory")
+	out, err := inspectField("test1", "HostConfig.KernelMemory")
 	c.Assert(err, check.IsNil)
 	if out != "52428800" {
 		c.Fatalf("setting the kernel memory limit failed")
 	}
+
+	dockerCmd(c, "run", "--kernel-memory", "100000K", "--name", "test2", "busybox", "true")
+
+	out, err = inspectField("test2", "HostConfig.KernelMemory")
+	c.Assert(err, check.IsNil)
+	if out != "102400000" {
+		c.Fatalf("setting the kernel memory limit failed")
+	}
+
+	dockerCmd(c, "run", "--kernel-memory", "1G", "--name", "test3", "busybox", "true")
+
+	out, err = inspectField("test3", "HostConfig.KernelMemory")
+	c.Assert(err, check.IsNil)
+	if out != "1073741824" {
+		c.Fatalf("setting the kernel memory limit failed")
```

Comment:
could you provide expected and actual values here so it'll be easier to see which failed?